src/main.rs

```diff
@@ -24,6 +24,20 @@ use {
     esp_idf_sys::esp_wifi_set_ps,
 };
 
+macro_rules! generate_register_modules {
```

Comment:
I'm not 100% sure if this macro belongs here, or if it should be exported from the micro-rdk. I think it probably belongs here because currently this whole auto-load protocol is more or less established between the current `micro-rdk-robot-template` and a proposed `micro-rdk-module-template`.

Reply:
we'll leave it here for now, but might make sense to have it live in the micro_rdk so others could use it

Reply:
Sounds good to me.